### PR TITLE
Integration with Snyk library endpoint

### PR DESCRIPTION
**Pull request by bmvermeer**
I created a vulnerability badge based on an endpoint I created for Snyk.
I deliberately do the calculation in the library-hero template to prevent slow rendering. So regardless of the Snyk endpoint being slow or not, the user sees the rendering.
I favor this over incorporating the data in the library property in the index.vue

### CODE REVIEW

File: components/library_hero.vue [hunk 1]
```diff
@@ -53,18 +59,38 @@
...
+        data () {
+            return {
+                vulns: null,
+            };
+        },
+        watch: {
+            version: {
+                immediate: true,
+                async handler () {
+                    this.vulns = await getVulns(this.$props.library, this.$props.version);
```
Comment (MattIPv4) [inline components/library_hero.vue:88]:
```suggestion
                    this.$data.vulns = await getVulns(this.$props.library, this.$props.version);
```

I would prefer to use `$data` here to be explicit?

File: components/library_hero.vue [hunk 2]
```diff
@@ -53,18 +59,38 @@
...
+                },
+            },
+        },
+        async created () {
+            this.vulns = await getVulns(this.$props.library, this.$props.version);
```
Comment (MattIPv4) [inline components/library_hero.vue:93]:
```suggestion
            this.$data.vulns = await getVulns(this.$props.library, this.$props.version);
```

Same here, I would prefer to use `$data` here to be explicit?

File: util/get_vulns.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import fetch from 'node-fetch';
+const snykApi = 'http://snyk-widget.herokuapp.com';
+const key = 'VRHr?YP_Gb7d9LFh$ugktZHUe+gGBp#LGG3*bDLt';
```
Comment (MattIPv4) [inline util/get_vulns.js:3]: Could these be moved into `data/config.js`?

Comment (MattIPv4) [inline util/get_vulns.js:3]: Is this key safe to be public?

File: util/get_vulns.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+export default async (library, version) => {
+    // await new Promise(r => setTimeout(r, 10000));
```
Comment (MattIPv4) [inline util/get_vulns.js:6]:
```suggestion
```
I'm guessing this was from testing?

File: components/library_hero.vue (hunk 2 above)
Comment (MattIPv4) [inline components/library_hero.vue:93]:
```suggestion
            this.vulns = await getVulns(this.$props.library, this.$props.version).catch(() => {});
```

This call failing will cause Nuxt to show the error page -- we can just add a .catch() on the end, so it is undefined it it fails?

File: components/library_hero.vue (hunk 1 above)
Comment (MattIPv4) [inline components/library_hero.vue:88]:
```suggestion
                    this.vulns = await getVulns(this.$props.library, this.$props.version).catch(() => {});
```

This call failing will cause Nuxt to show the error page -- we can just add a .catch() on the end, so it is undefined it it fails?

File: util/get_vulns.js (hunk 3 above)
Comment (bmvermeer) [reply util/get_vulns.js:3]: Yes that would be better :)

Comment (bmvermeer) [reply util/get_vulns.js:3]: Yes, the data is open. However, if we see someone is overusing it we can rate limit. That's why we hand out separate keys

File: util/get_vulns.js (hunk 4 above)
Comment (bmvermeer) [reply util/get_vulns.js:6]: oops, yes

**Comment by bmvermeer** [comment]
Thanks for the feedback, I will work it out in a few hours.

**Pull request merged**